### PR TITLE
Run mypy during CI, ignore most existing errors (GEA-11717)

### PR DESCRIPTION
**Pull request by kenany**
mypy has been a dev dependency for a while but it wasn't running during CI. So, type-check errors have been building up over time. There are over a 100 errors, many of which would require some refactoring to fix, so in order to at least have type-checking for all code going further I've gone and `# type: ignore[*]`'d most of the existing issues. Over time, as we work on various parts of the project, we can work to remove the ignores and introduce some of the stricter type-check options.

### CODE REVIEW

File: packages/pangea-sdk/.sdk-ci.yml [hunk 1]
```diff
@@ -35,6 +35,13 @@ generate_docs:
     paths: ['packages/pangea-sdk/python_sdk.json']
     when: on_success
 
+mypy:
```
Comment (doubletooth) [inline packages/pangea-sdk/.sdk-ci.yml:38]: nit: I'd call this job `type_check` and the one above `lint`

File: packages/pangea-sdk/.sdk-ci.yml [hunk 2]
```diff
@@ -35,6 +35,13 @@ generate_docs:
...
+  extends: .pangea-sdk-base
+  stage: lint
+  needs: []
+  script:
+    - poetry run mypy pangea tests
```
Comment (doubletooth) [inline packages/pangea-sdk/.sdk-ci.yml:43]: what's the image being used in tests?

Comment (kenany) [reply packages/pangea-sdk/.sdk-ci.yml:43]: Should be `python:3.7-bullseye` (from root `.gitlab-ci.yml`).

File: packages/pangea-sdk/.sdk-ci.yml (hunk 1 above)
Comment (kenany) [reply packages/pangea-sdk/.sdk-ci.yml]: Done.

**Review by pangea-andrest — COMMENTED** [review]
LGTM

**Review by doubletooth — APPROVED** [review]

**Comment by kenany** [comment]
We should've let me squash my `fixup!` commits before merging this.

**Comment by doubletooth** [comment]
Apologies @kenany. Do you prefer I just approve and leave it at that?

**Comment by kenany** [comment]
"Squad and merge" button would've sufficed, that's what I would've done here anyways.

**Comment by doubletooth** [comment]
Will do next time.

**Pull request merged**